flocker/ca/_script.py

```diff
@@ -90,6 +90,68 @@ def getSynopsis(self):
 
 
 @flocker_standard_options
+class UserCertificateOptions(PrettyOptions):
+    """
+    Command line options for ``flocker-ca create-api-certificate``.
+    """
+
+    helptext = """Create a new certificate for an API end user.
+
+    Creates a certificate signed by a previously generated certificate
+    authority (see flocker-ca initialize command for more information).
+
+    Required parameters:
+
+    * name: A username for which the certificate should be created.
+    """
+
+    synopsis = "<name> [options]"
+
+    optParameters = [
+        ['inputpath', 'i', None,
+         ('Path to directory containing root certificate.'
+          'Defaults to current working directory.')],
+        ['outputpath', 'o', None,
+         ('Path to directory to write control service certificate.'
+          'Defaults to current working directory.')],
+    ]
+
+    def parseArgs(self, name):
+        self["name"] = name
+
+    def run(self):
+        """
+        Create a new node certificate signed by the root and write it out to
+        the current directory.
+
+        :raise PathError: When the root certificate and key cannot be found.
+        """
+        if self["inputpath"] is None:
+            self["inputpath"] = os.getcwd()
+        if self["outputpath"] is None:
+            self["outputpath"] = os.getcwd()
+
+        self["inputpath"] = FilePath(self["inputpath"])
+        self["outputpath"] = FilePath(self["outputpath"])
+
+        try:
+            try:
+                ca = RootCredential.from_path(self["inputpath"])
+                uc = UserCredential.initialize(
+                    self["outputpath"], ca, self["name"])
+                self._sys_module.stdout.write(
+                    b"Created {user}.crt. You can now give it to your "
+                    "API enduser so they can access the control service "
+                    "API.".format(user=uc.username)
```

Comment:
Formatting unicode into bytes is not a good idea.  Format it into a unicode template string and the encode the entire thing using an encoding appropriate for the output stream (`stdout.encoding`, probably).